### PR TITLE
properly handle biri.Error

### PR DESCRIPTION
**Pull request by arnitolog**
fixes https://github.com/warpstreamlabs/bento/issues/162

### CODE REVIEW

**Review by gregfurman — COMMENTED** [review]
Hey! Thanks for the contribution and quick fix for this. Any chance you have capacity to add an integration test to the `opensearch` suite for this case?

There are 2 deletion tests but none test this case you've encountered of trying to delete a non-existent document AFAIK. See [opensearch/integration_test.go](https://github.com/warpstreamlabs/bento/blob/main/internal/impl/opensearch/integration_test.go#L432).

**Comment by arnitolog** [comment]
@gregfurman, I've added integration test for this case. Please take a look

File: internal/impl/opensearch/integration_test.go [hunk 1]
```diff
@@ -451,6 +451,41 @@ action: ${! @elastic_action }
...
+
+	// Verify the document was not found
+	get, err := client.Do(ctx, osapi.DocumentGetReq{
+		Index:      "test_conn_index",
+		DocumentID: "non-existing-id",
+	}, nil)
+	require.NoError(t, err)
+	if get.IsError() {
+		var respCode int
+		respCode = get.StatusCode
```
Comment (gregfurman) [inline internal/impl/opensearch/integration_test.go:480]: Linter doesn't like this one.
```suggestion
		respCode := get.StatusCode
```

**Comment by gregfurman** [comment]
Awesome! Thanks for adding that test. 

Your golangci-lint step is failing so just run `make fmt && make lint` and happy to approve 😄 

I added a suggestion that you can also accept that should fix it.

**Comment by arnitolog** [comment]
@gregfurman , fixed

**Review by gregfurman — APPROVED** [review]
LGTM! Thanks for this 👍

**Pull request merged**